server/config/store_config.go

```diff
@@ -118,6 +119,32 @@ func (c *StoreConfig) GetRegionMaxKeys() uint64 {
 	return uint64(c.Coprocessor.RegionMaxKeys)
 }
 
+// CheckRegionSize return error if the smallest region's size is less than mergeSize
+func (c *StoreConfig) CheckRegionSize(size, mergeSize uint64) error {
+	if size < c.GetRegionMaxSize() {
+		return nil
+	}
+
+	if smallSize := size % c.GetRegionSplitSize(); smallSize <= mergeSize && smallSize != 0 {
+		log.Info("region size is too small", zap.Uint64("size", size), zap.Uint64("mergeSize", mergeSize), zap.Uint64("smallSize", smallSize))
+		return errs.ErrCheckerMergeAgain.FastGenByArgs("the smallest region of the split regions is less than max-merge-region-size, " +
```

Comment:
Maybe preventing it from splitting is not always a good way. BTW, we have `split-merge-interval` to make sure the split region won't be merged again within a fixed time interval. If we introduce this mechanism, the `split-merge-interval` will be less useful?

Reply:
In this case ,  the `split-merge-interval`  mechanism  only prevent from merge new region,  but the region will be merged again after `split-merge-interval` seconds ago.

Reply:
Oh, I misunderstand the meaning. Better to add a simple example to explain why we do it.